### PR TITLE
Add route params management

### PR DESCRIPTION
**Pull request by MarieMinasyan**
Adds route params management in the menu item extension

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | `-`
| Added Behats                      | `-`
| Added integration tests           | `-`
| Changelog updated                 | `-`
| Review and 2 GTM                  | `-`
| Micro Demo to the PO (Story only) | `-`
| Migration script                  | `-`
| Tech Doc                          | `-`

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/menu/item.js [hunk 1]
```diff
@@ -93,6 +93,15 @@ define(
             },
 
             /**
+             * Returns the route parameters.
+             *
+             * @returns {json}
```
Comment (tamarasaurus) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/menu/item.js]: should be {Object}

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/menu/column.js [hunk 2]
```diff
@@ -88,8 +88,9 @@ define(
 
                 this.getRoot().trigger('pim_menu:register_item', {
                     target: this.getTab(),
-                    route: navigationItem.code,
-                    position: navigationItem.position
+                    route: navigationItem.route,
```
Comment (jmleroux) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/menu/column.js]: :+1:

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/menu/column.js [hunk 3]
```diff
@@ -88,8 +88,9 @@ define(
...
+                    position: navigationItem.position,
+                    routeParams: navigationItem.routeParams
```
Comment (jmleroux) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/menu/column.js]: Do we use camelCase ou snake_case (`route_params`) in YML?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/menu/item.js [hunk 4]
```diff
@@ -50,9 +50,10 @@ define(
              */
             configure: function () {
                 this.trigger('pim_menu:column:register_navigation_item', {
-                    code: this.getRoute(),
+                    route: this.getRoute(),
```
Comment (jmleroux) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/menu/item.js]: :+1:

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/menu/item.js [hunk 5]
```diff
@@ -93,6 +94,15 @@ define(
             },
 
             /**
+             * Returns the route parameters.
+             *
+             * @returns {Object}
+             */
+            getRouteParams: function () {
+                return this.config.routeParams !== 'undefined' ? this.config.routeParams : {};
```
Comment (jmleroux) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/menu/item.js]: Same question about camelCase vs snake_case

**Comment by jmleroux** [comment]
We have to run tests, becasue i think there are a lot of broken config files because of the change `code` => `route`.

_but it's for the win_

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/menu/column.js (hunk 3 above)
Comment (phaseinducer) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/menu/column.js]: camel case is ok

**Pull request merged**